OMCompiler/Compiler/BackEnd/Initialization.mo

```diff
@@ -357,10 +358,14 @@ protected function inlineWhenForInitialization "author: lochel
   output BackendDAE.BackendDAE outDAE = inDAE;
 protected
   list<BackendDAE.Equation> eqnlst;
+  list<BackendDAE.Equation> clockEqnsLst;
   HashSet.HashSet leftCrs = HashSet.emptyHashSet() "dummy hash set - should always be empty";
 algorithm
   outDAE.eqs := List.map(inDAE.eqs, inlineWhenForInitializationSystem);
   (eqnlst, _) := BackendEquation.traverseEquationArray(inDAE.shared.removedEqs, inlineWhenForInitializationEquation, ({}, leftCrs));
+  // TODO AHEU: Add simCodeTarget C around this?
+  clockEqnsLst := BackendEquation.traverseEquationArray(inDAE.shared.removedEqs, SynchronousFeatures.getBoolClockWhenClauses, {});
+  eqnlst := listAppend(clockEqnsLst, eqnlst);
```

Comment:
This is for models that use event clocks with a condition that becomes true at initialization, so the first tick needs to be done at simulation time `t=0` (after initialization).
We check if the condition becomes true, but because it became true at initialization the clock will not activate at the first simulation time point `t=0`.

I'm not really happy with my solution and I think that adding the when-equation to the initial system is not the best solution. It probably also affects the C++ runtime, so I need to check for `simCodeTarget`.

Reply:
I'm not sure how this worked before and can't find the model with this problem.